File: 0738-monotone-increasing-digits/0738-monotone-increasing-digits.js
/**
 * @param {number} n
 * @return {number}
 */
var monotoneIncreasingDigits = function(n) {
    let s = n.toString().split('');
    let make = 0;
    for(let i = s.length-1; i >= 1; --i) {
        if(s[i] < s[i-1]) {
            make = i;
            s[i-1] = (parseInt(s[i-1])-1).toString();
        }
    }
    if(make == 0) return n;
    for(let i = make; i < s.length; ++i)
        s[i] = '9';
    return parseInt(s.join(''));
};

File: 0738-monotone-increasing-digits/0738-monotone-increasing-digits.cpp
class Solution {
public:
    int monotoneIncreasingDigits(int n) {
        string s = to_string(n);
        int make = 0;
        for(int i = s.length()-1; i >= 1; --i) {
            if(s[i] < s[i-1]) {
                make = i;
                --s[i-1];
            }
        }
        
        if(make == 0) return n;
        
        for(int i = make; i < s.length(); ++i)
            s[i] = '9';
        
        return stoi(s);
    }
};